syntax = "proto3";

service Backend {
  rpc info (Void) returns (Info) {}
  rpc state (Void) returns (State) {}
  rpc notifications (Void) returns (NotificationList) {}
}

message Void {}

enum Speed {
  STOP = 0;
  SLOW = 1;
  MEDIUM = 2;
  FAST = 3;
}

message Info {
  string project = 1;
  string version = 2;
  string description = 3;
  string author = 4;
  string contact = 5;
  string license = 6;
  float time = 7;
  string commit = 8;
}

message State {
  string name = 1;
  bool is_locked = 2;
  bool can_drive_auto = 3;
  bool can_drive_manual = 4;
  bool is_fatal = 5;
  bool is_warn = 6;
  Speed speed = 7;
}

enum Level {
  INFO = 0;
  WARNING = 1;
  FATAL = 2;
}

message Notification {
  int32 id = 1;
  Level level = 2;
  string title = 3;
  string description = 4;
  string location = 5;
  float start_time = 6;
  string errorrnr = 7;
  repeated string possible_sources = 8;
}

message NotificationList {
  repeated Notification notifications = 1;
}
